lib/model.py

```diff
@@ -36,39 +39,74 @@ def SetupModel(self):
         if self.CONFIG['CKPT']['TURN_ON']:
             self.load_checkpoint()
 
-        self.set_dataset()
-        self.set_data_iterator()
-        self.set_validation()
+        self.devide_datasets()
+        self.set_datasets()
         self.set_loss_collector()
 
         if self.CONFIG['BASE']['IS_MASTER']:
             print(f"Model {self.CONFIG['BASE']['MODEL_ID']} has successively created")
 
-    def load_next_batch(self):
+    def devide_datasets(self):
```

Comment:
Users should be less likely to modify the lib/dataset.py
Make this function independent to data type.